File: LeetCode/Simulation/JS/Leetcode1742-maximum-number-of-balls-in-a-box.js
var countBalls = function (lowLimit, highLimit) {
  let [ans, mp] = [0, new Map()];
  let calcu = (x) => {
    let res = 0;
    while (x) {
      res += x % 10;
      x = parseInt(x / 10);
    }
    return res;
  }
  for (let i = lowLimit; i <= highLimit; i++) {
    mp.set(calcu(i), (mp.get(calcu(i)) | 0) + 1);
  }
  for (const x of mp.values()) {
    ans = Math.max(ans, x);
  }
  return ans;
};

File: LeetCode/Simulation/Golang/Leetcode1742-maximum-number-of-balls-in-a-box.go
package sbgo

func countBalls(lowLimit int, highLimit int) int {
	ans, mp := 0, make(map[int]int)
	calcu := func(x int) int {
		res := 0
		for x != 0 {
			res += x % 10
			x /= 10
		}
		return res
	}
	for i := lowLimit; i <= highLimit; i++ {
		mp[calcu(i)]++
	}
	max := func(i, j int) int {
		if i < j {
			return j
		}
		return i
	}
	for _, item := range mp {
		ans = max(ans, item)
	}
	return ans
}
